lib/matplotlib/tests/test_datetime.py

```diff
@@ -32,11 +32,27 @@ def test_axhline(self):
         fig, ax = plt.subplots()
         ax.axhline(...)
 
-    @pytest.mark.xfail(reason="Test for axhspan not written yet")
     @mpl.style.context("default")
     def test_axhspan(self):
-        fig, ax = plt.subplots()
-        ax.axhspan(...)
+        mpl.rcParams["date.converter"] = 'concise'
+        np.random.seed(19680801)
+
+        start_date = datetime.datetime(2023, 1, 1)
+        time_delta = datetime.timedelta(days=1)
+
+        fig, (ax1, ax2) = plt.subplots(2, 1, constrained_layout=True, figsize=(10, 8))
+
+        ax1.set_ylim(start_date, start_date + 29*time_delta)
+        for i in range(np.random.randint(1, 5)):
```

Comment:
Could you take out all the random draws? It's easier to interpret correctness of a test image with hardcoded values.